### ISSUE TITLE
Theme download gives 404 Not Found

### ISSUE THREAD

**Issue by lukehatfield** [Shopify/themekit#523]
### What happened
When I run `theme download -e=2018v1` I get a 404 Not Found error.
````
[2018v1]: 212 / 214 [==================================================]  99 %
[2018v1] error downloading asset: Asset Perform Retrieve to  at host xxx.myshopify.com
    Status: 404 Not Found
[2018v1]: 213 / 214 [==================================================] 100 %
[2018v1] error writing asset: open /Users/luke/path/xxx.com/code[2018v1]: 214 / 214 [==================================================] 100 %
````

### What I expected to happen
I hoped that all files from that theme would be downloaded, according to the theme.lock file.

### Steps to reproduce

1. Run `theme download -e=2018v1` from git project folder.

### My Environment

OS (including version): macOS High Sierra, 10.13.5
Themekit version: ThemeKit 0.7.5 darwin/amd64
Ignore Files(if any): none

**Comment by eddyparkinson** [comment]
Me too, I think.  

Fix
---
 Please add the filepath/filename to 404 errors.

Error  
---

When I download a folder. I sometimes get 404 errors.  

Cause
---

The error is caused by a file in the folder being wrong - (renamed, edited etc). If I do each file one by one I can find the problem file and fix the problem.  

The error is because of a file being out of sync. Such as renamed or edited. I suspect file date and name is being checked, and any miss match gives a 404 error. But it is not showing what file causes the error.

Related error
---

templates/customers  is a folder and so gives an error - would be good if it could ignore folders


My Environment
----
 ThemeKit 0.7.5 windows/amd64
 Windows 10

**Comment by tanema** [comment]
Thank you for the investigation! However, a few questions. Themekit already does not request folders, are you seeing an error with that folder name specifically in the requested action?

Second the filename is missing which is why there is a 404 and that is why there is a blank in the error message. I have not yet narrowed down why that is.

**Comment by lukehatfield** [comment]
I found that if I rename my environment to `2018v.1` or `2018ver1` or just add one more character after the v, it will work. What didn't work was `2018v1`. Don't know if there is a limit on how many characters the env name needs to be or the balance of letters and numbers? Either way, seems to be working again for me.

**Comment by eddyparkinson** [comment]
Tanema, Thanks for explaining, that helps.

>Themekit already does not request folders:

Like this:

```
$ theme download -e=development templates/*
[development]: 20 / 21 [====================================>--]  95 %
[development] error downloading asset: Asset Perform Retrieve to  at host site.myshopify.com
        Status: 404 Not Found
[development]: 21 / 21 [=======================================] 100 %

```

> the filename is missing which is why there is a 404
Also, looks to happen with date sync. Not sure the error is 404, but some error and no indication of what file caused the error.

**Comment by tanema** [comment]
So when you run `theme download -e=development templates/*` themekit is not expanding the asterisk. Your command line expands it, so the problem is that you are basically running `theme download -e=development templates/customers` This may be a clue why you are getting that error though because if you try to download `templates/customers` directly, you will get a 404 and themekit will not be able to identify it within the project either. This seems the be an issue with passing wildcards into themekit

```
➜  theme download -e=development templates/customers
[development]: 0 / 1 [--------------------------------------------------------------------]   0 %
[development] error downloading asset: Asset Perform Retrieve to  at host can-i-buy-a-feeling.myshopify.com
        Status: 404 Not Found
[development]: 1 / 1 [====================================================================] 100 %
```

**Comment by tanema** [comment]
This has been fixed in the v0.8.0 release. It will now handle wildcard flags and directories a lot more gracefully.
